### PR TITLE
TraceableCacheAdapter compatible with Symfony\Component\Cache\Adapter…

### PR DESCRIPTION
**Pull request by jaras2678**
Issuse https://github.com/getsentry/sentry-symfony/issues/576

### CODE REVIEW

**Comment by ste93cry** [comment]
This change is a breaking change. Moreover, it doesn't work on PHP 7 because the `mixed` typehint was introduced in version 8. There's already an open PR to support Symfony 6, I will find a way to integrate this change there

File: src/Tracing/Cache/TraceableCacheAdapterTrait.php [hunk 1]
```diff
@@ -35,7 +36,7 @@ trait TraceableCacheAdapterTrait
     /**
      * {@inheritdoc}
      */
-    public function getItem($key)
+    public function getItem(mixed $key): CacheItem
```
Comment (Jean85) [inline src/Tracing/Cache/TraceableCacheAdapterTrait.php:39]: The variance rules are reversed for arguments, so just by doing this we should be fine:
```suggestion
    public function getItem($key): CacheItem
```

**Review by Jean85 — CHANGES_REQUESTED** [review]
Also, please add a changelog entry. I would also require a test, but I would expect this to land from #566. @ste93cry can you check on this?

**Comment by ste93cry** [comment]
@Jean85 there's no need to do anything here, as I said before there's already an open PR to support Symfony 6, with this change included

**Pull request closed without merging**